tools/readme_linter/assert.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+
+	"github.com/yuin/goldmark/ast"
+)
+
+//type for all linter assert methods
+type T struct {
+	filename       string
+	markdown       []byte
+	newlineOffsets []int
+
+	fails int
+}
+
+// called by all assert functions that involve a node
+func (t *T) printFailedAssertf(n ast.Node, format string, args ...interface{}) {
+	t.printFile(n)
+	fmt.Printf(format+"\n", args...)
+	//t.printRule(3)
+	t.fails++
+}
+
+// Assert function that doesnt involve a node, for example if something is missing
+func (t *T) assertf(format string, args ...interface{}) {
+	t.assertLinef(0, format, args...) // There's no line number associated, so use the first
+}
+
+func (t *T) assertNodef(n ast.Node, format string, args ...interface{}) {
+	t.printFailedAssertf(n, format, args...)
+}
+
+func (t *T) assertLinef(line int, format string, args ...interface{}) {
+	t.printFileLine(line)
+	fmt.Printf(format+"\n", args...)
+	//t.printRule(2) This can either be 2 or 3 since it's called through assertf
+	t.fails++
+}
+
+// func (t *T) printRule(callers int) {
+// 	pc, codeFilename, codeLine, ok := runtime.Caller(callers)
+// 	if !ok {
+// 		panic("can't get caller")
+// 	}
+
+// 	f := runtime.FuncForPC(pc)
+// 	var funcName string
+// 	if f != nil {
+// 		funcName = f.Name()
+// 	}
+
+// 	fmt.Printf("%s:%d: ", codeFilename, codeLine)
+// 	if len(funcName) == 0 {
+// 		fmt.Printf("failed assert\n")
+// 	} else {
+// 		fmt.Printf("failed assert in function %s\n", funcName)
+// 	}
+// }
```

Comment:
Same as above, if we are going to leave this commented out, can you leave a comment saying why for the next person to understand why.

Reply:
It's uncommented now. See the comment on printFailedAssertf